servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -104,6 +106,134 @@
 
   /** The service that tries to check inodefiles with ttl set */
   private Future<?> mTTLCheckerService;
+  /** The configuration of {@link Constants#MASTER_TTLCHECKER_INTERVAL_MS} */
+  private int mTTLCheckerIntervalMs;
+
+  /**
+   * A bucket with all files whose ttl value lies in the bucket's time interval. The bucket's time
+   * interval starts at a specific time and lasts for {@link #mTTLCheckerIntervalMs}.
+   *
+   * Not thread-safe. Only for use related to {@link TTLBucketList}.
+   */
+  private class TTLBucket {
+    /**
+     * Each bucket has a time to live interval, this value is the start of the interval, interval
+     * value is the same as the configuration of {@link Constants#MASTER_TTLCHECKER_INTERVAL_MS}.
+     */
+    private long mTTLIntervalStartTimeMs;
+    /** A list of InodeFiles whose ttl value is in the range of this bucket's interval. */
+    private List<InodeFile> mFiles;
+
+    public TTLBucket(long startTimeMs) {
+      mTTLIntervalStartTimeMs = startTimeMs;
+      mFiles = new LinkedList<InodeFile>();
+    }
+
+    public long getTTLIntervalStartTimeMs() {
+      return mTTLIntervalStartTimeMs;
+    }
+
+    public void setTTLIntervalStartMs(long startTimeMs) {
+      mTTLIntervalStartTimeMs = startTimeMs;
+    }
+
+    public long getTTLIntervalEndTimeMs() {
+      return mTTLIntervalStartTimeMs + mTTLCheckerIntervalMs;
+    }
+
+    public List<InodeFile> getFiles() {
+      return mFiles;
+    }
+
+    public void addFile(InodeFile file) {
+      mFiles.add(file);
+    }
+  }
+
+  /**
+   * A list of non-empty {@link TTLBucket}s sorted by ttl interval start time of each bucket.
+   *
+   * Two adjacent buckets may not have adjacent intervals since there may be no files with ttl value
+   * in the skipped intervals.
+   *
+   * Thread-safety is guaranteed by {@link ConcurrentSkipListSet}.
+   */
+  private class TTLBucketList {
+    /**
+     * List of buckets sorted by interval start time.
+     * SkipList is used for O(logn) insertion and retrieval, see {@link ConcurrentSkipListSet}.
+     */
+    private ConcurrentSkipListSet<TTLBucket> mBucketList;
+
+    public TTLBucketList() {
+      mBucketList = new ConcurrentSkipListSet<TTLBucket>(new Comparator<TTLBucket>() {
```

Comment:
Does it make more sense to have TTLBucket implement comparable rather than passing in this comparator.